pkg/controller/controller_instance.go

```diff
@@ -768,16 +776,17 @@ func (c *controller) resolveReferences(instance *v1beta1.ServiceInstance) (*v1be
 			var scErr error
 			sc, scErr = c.serviceClassLister.Get(instance.Spec.ClusterServiceClassRef.Name)
 			if scErr != nil {
-				return nil, fmt.Errorf(`Couldn't find ClusterServiceClass (K8S: %s)": %v`, instance.Spec.ClusterServiceClassRef.Name, scErr.Error())
+				return nil, false, fmt.Errorf(`Couldn't find ClusterServiceClass (K8S: %s)": %v`, instance.Spec.ClusterServiceClassRef.Name, scErr.Error())
 			}
 		}
 
 		instance, err = c.resolveClusterServicePlanRef(instance, sc.Spec.ClusterServiceBrokerName)
 		if err != nil {
-			return nil, err
+			return nil, false, err
 		}
 	}
-	return c.updateServiceInstanceReferences(instance)
+	updatedInstance, err := c.updateServiceInstanceReferences(instance)
+	return updatedInstance, true, err
```

Comment:
Shall we return true only if the update has succeeded? i.e.
```go
return updatedInstance, err == nil, err
```

Reply:
Yes, absolutely. Missed that because I introduced the bool at the last minute (previously, I had the method simply return `nil, nil` to signal that it hadn't modified the instance; afters speaking to Paul about that, I introduced the bool to make the method more explicit).